prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTask.java

```diff
@@ -0,0 +1,171 @@
+			String kindPluralName = getRequiredParameterValue(workContext, "kind-plural-name");
+			String operation = getRequiredParameterValue(workContext, "operation");
+
+			OperationType operationType = OperationType.valueOf(operation.toUpperCase());
+
+			switch (operationType) {
+				case UPDATE:
+					update(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case CREATE:
+					create(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case GET:
+					get(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+			}
+		}
+		catch (MissingParameterException | IOException | ApiException | IllegalArgumentException e) {
+			log.error("Kubeapi task failed", e);
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+	}
+
+	private void get(WorkContext ctx, String apiGroup, String apiVersion, String kindPluralName)
+			throws MissingParameterException, ApiException {
+		String resourceName = getRequiredParameterValue(ctx, "resource-name");
+		String resourceNamespace = getRequiredParameterValue(ctx, "resource-namespace");
+		String workCtxKey = getRequiredParameterValue(ctx, "work-ctx-key");
+
+		DynamicKubernetesObject obj = api.get(apiClient, apiGroup, apiVersion, kindPluralName, resourceNamespace,
+				resourceName);
+		String resourceJson = obj.getRaw().toString();
+		ctx.put(workCtxKey, resourceJson);
+	}
+
+	private void create(WorkContext ctx, String apiGroup, String apiVersion, String kindPluralName)
+			throws MissingParameterException, ApiException {
+		String resourceJson = getRequiredParameterValue(ctx, "resource-json");
+		DynamicKubernetesObject obj = Dynamics.newFromJson(resourceJson);
+		api.create(apiClient, apiGroup, apiVersion, kindPluralName, obj);
+	}
+
+	private void update(WorkContext ctx, String apiGroup, String apiVersion, String kindPluralName)
+			throws MissingParameterException, ApiException {
+		String resourceJson = getRequiredParameterValue(ctx, "resource-json");
+		DynamicKubernetesObject newObj = Dynamics.newFromJson(resourceJson);
+		String resourceNamespace = newObj.getMetadata().getNamespace();
+		String resourceName = newObj.getMetadata().getName();
+		DynamicKubernetesObject currObj = api.get(apiClient, apiGroup, apiVersion, kindPluralName, resourceNamespace,
+				resourceName);
+		V1ObjectMeta metadata = newObj.getMetadata();
+		metadata.setResourceVersion(currObj.getMetadata().getResourceVersion());
+		newObj.setMetadata(metadata);
+		api.update(apiClient, apiGroup, apiVersion, kindPluralName, newObj);
+	}
+
+	private void initClient() throws IOException {
+		if (apiClient != null) {
+			return;
+		}
+		createClientInstance();
+	}
+
+	private synchronized void createClientInstance() throws IOException {
+		if (apiClient != null) {
+			return;
+		}
+		this.apiClient = ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(new StringReader(kubeconfig))).build();
```

Comment:
@pkliczewski is this workflow type expected to use a [user account or a service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#user-accounts-versus-service-accounts) to interact with the cluster? For application processes, it's recommended to use a service account as it makes it easier to automate and has less security risks compared to using a kubeconfig derived from a user account.

Also, do we plan to allow Parodos kube tasks to reach out to other clusters? I suspect that in general  we should first check for a token + ca + url in the parameters when we want to interact remotely with a different cluster, and if not available default to retrieving the credentials from the same cluster where Parodos is running:
1) load $KUBECONFIG
2) load from $HOME/.kube/config
3) load from in-cluster service account if available

Like it's being done [here](https://github.com/kubernetes-client/java/blob/7c00056d11b80aa94fe7ccb711db3ea4f1abf06d/util/src/main/java/io/kubernetes/client/util/Config.java#L102-L110).

Reply:
this task is for interacting with a cluster anywhere in the world. not the cluster where parodos is running. after all, parodos does not have to be in a cluster. allowing to specify a kubeconfig location enables the user to have multiple tasks with different kubeconfigs. Even for the same cluster.

Reply:
how the kubeconfig is configured (service account, user account, token password etc) is 100 up to the user and we do not limit.